courses/fundamentals_of_ada/070_subprograms.rst

```diff
@@ -445,162 +392,62 @@ Why Read Mode `out` Parameters?
      end loop;
    end Compute;
 
----------------------------------
-Modes' Requirements for Actuals
----------------------------------
-
-* Modes :ada:`in out` and :ada:`out`
-
-   - Must be a **writable object**
-   - **No** expressions
-
-* Mode :ada:`in`
-
-   - May use expressions (actual can't be altered)
-
-.. code:: Ada
-
-   procedure Do_Something (X : in     Integer;
-                           Y :    out Integer );
-   ...
-   begin
-     Do_Something(X + 2, Y); -- legal
-     Do_Something(X, Y + 1); -- compile error
-
--------------------------------------
-Parameter Defaults May Be Specified
--------------------------------------
-
-* :ada:`in` parameters only
-* Default used when **no value** is provided
-
-.. code:: Ada
-
-   My_Process : Process_Name;
-   Before, After : Duration;
-   procedure Activate( Process : in Process_Name;
-                       Before : in Duration := 0.0;
-                       After : in Duration := 0.0 );
-   ...
-   begin
-     -- no defaults taken
-     Activate (My_Process, Before, After);
-     -- defaults for Before, After
-     Activate (My_Process);
-     -- defaults for After
-     Activate (My_Process, 60.0);
-
----------------------------------
-Skipping Over Actual Parameters
----------------------------------
-
-* Requires named format for remaining arguments
-
-.. code:: Ada
-
-   procedure Activate (
-     Process : in Process_Name := Unknown;
-     Before : in Duration := 0.0;
-     After : in Duration := 0.0);
-   ...
-   begin
-     -- Parameter "Process" is skipped
-     Activate (Before => 60.0, After => 10.0);
-     Activate (60.0, 60.0); -- compile error
-     -- Parameter "Before" is skipped
-     Activate (My_Process, After => 10.0);
-     Activate (My_Process, 60.0); -- Not an error, but WRONG!
-
-.. container:: speakernote
-
-   Not using named association can cause confusion if future development adds parameters
-
 ------------------------------
 Parameter Passing Mechanisms
 ------------------------------
 
-* Passed either "by-copy" or "by-reference"
 * By-Copy
 
    - The formal denotes a separate object from the actual
-   - A copy of the actual is placed into the formal before the call
-   - A copy of the formal is placed back into the actual after the call
+   - :ada:`in`, :ada:`in out`: actual is copied into the formal **before** the call
```

Comment:
Would it be better to say "on entry to" and "on exit from" the subprogram?

Reply:
:+1: